core/vm/evm.go

```diff
@@ -47,14 +47,40 @@ type (
 
 // run runs the given contract and takes care of running precompiles with a fallback to the byte code interpreter.
 func run(evm *EVM, contract *Contract, input []byte, readOnly bool) ([]byte, error) {
-	// Intercept the StateManager calls
-	if contract.Address() == StateManagerAddress {
-		log.Debug("Calling State Manager contract.", "StateManagerAddress", StateManagerAddress.Hex())
-		ret, err := callStateManager(input, evm, contract)
-		if err != nil {
-			log.Error("State manager error!", "error", err)
+	if UsingOVM {
+		// OVM_ENABLED
+
+		// Some simple logging here. First, check to see if we know about the address we're
+		// interacting with and try to log the input data.
+		var isUnknown = true
+		for name, account := range OvmStateDump.Accounts {
+			if contract.Address() == account.Address {
+				isUnknown = false
+				abi := &(account.ABI)
+				method, err := abi.MethodById(input)
+				if err != nil {
+					log.Debug("Calling Known Contract", "Name", name, "ERROR", err)
+				} else {
+					log.Debug("Calling Known Contract", "Name", name, "Method", method.RawName)
+				}
+			}
+		}
```

Comment:
`OvmStateDump.Accounts` is already a [map](https://github.com/ethereum-optimism/go-ethereum/blob/dd0caa08486314f00c2422231e134ffc77d45d21/core/vm/ovm_state_dump.go#L3663), can't you just do `account, ok := OvmStateDumpAccounts[name]`?